src/statement-list/converters/expr2statement_list.cpp

```diff
@@ -25,3 +65,219 @@ std::string type2stl(const typet &type, const namespacet &ns)
   // CBMC types to STL code.
   return type2c(type, ns);
 }
+
+expr2stlt::expr2stlt(const namespacet &ns)
+  : inside_bit_string(false), is_reference(false), ns(ns)
+{
+}
+
+std::string expr2stlt::convert(const exprt &expr)
+{
+  if(ID_and == expr.id())
+    convert_and(expr);
```

Comment:
Since you already confirmed that it is an `and_exprt`, you can cast it here using `to_and_expr`. That also allows you to use overloading with the `convert` methods, i.e. you won't have to call them `convert_and`, but instead `convert(const and_exprt &`).